pkg/kubectl/cmd/apply.go

```diff
@@ -191,11 +233,112 @@ func RunApply(f *cmdutil.Factory, cmd *cobra.Command, out io.Writer, options *re
 	if err != nil {
 		return err
 	}
-
 	if count == 0 {
 		return fmt.Errorf("no objects passed to apply")
 	}
 
+	if !options.Prune {
+		return nil
+	}
+
+	selector, err := labels.Parse(options.Selector)
```

Comment:
:+1: for separate function, but I'd rather have it return err and check it here, than what you propose.